ada_feeding/ada_feeding/behaviors/transfer/compute_move_to_mouth_position.py

```diff
@@ -0,0 +1,159 @@
+        self,
+        face_detection_msg: Union[BlackboardKey, FaceDetection],
+        frame_id: Union[BlackboardKey, str] = "j2n6s200_link_base",
+        position_offset: Union[BlackboardKey, Tuple[float, float, float]] = (
+            0.0,
+            0.0,
+            0.0,
+        ),
+    ) -> None:
+        """
+        Blackboard Inputs
+
+        Parameters
+        ----------
+        face_detection_msg: The face detection message.
+        frame_id: The frame ID for the target position.
+        position_offset: The offset to add to the target position, in `frame_id`.
+        """
+        # pylint: disable=unused-argument, duplicate-code
+        # Arguments are handled generically in base class.
+        super().blackboard_inputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
+
+    def blackboard_outputs(
+        self,
+        target_position: Optional[
+            BlackboardKey
+            # PointStamped
+        ],
+    ) -> None:
+        """
+        Blackboard Outputs
+        By convention (to avoid collisions), avoid non-None default arguments.
+
+        Parameters
+        ----------
+        target_position: The target position to move the robot's end effector to
+            will be written to this key, as a PointStamped in `frame_id`.
+        """
+        # pylint: disable=unused-argument, duplicate-code
+        # Arguments are handled generically in base class.
+        super().blackboard_outputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
+
+    @override
+    def setup(self, **kwargs):
+        # Docstring copied from @override
+
+        # pylint: disable=attribute-defined-outside-init
+        # It is okay for attributes in behaviors to be
+        # defined in the setup / initialise functions.
+
+        # Get Node from Kwargs
+        self.node = kwargs["node"]
+
+        # Get TF Listener from blackboard
+        # For transform approach -> end_effector_frame
+        self.tf_buffer, _, self.tf_lock = get_tf_object(self.blackboard, self.node)
+
+    def update(self) -> py_trees.common.Status:
+        """
+        Computes the target position to move the robot's end effector to based on
+        the detected mouth center. Specifically, it transforms the detected mouth
+        center from the camera frame to the requested frame and then adds an offset.
+        It immediately returns either SUCCESS or FAILURE, and never returns RUNNING.
+        """
+        self.logger.info(f"{self.name} [ComputeMoveToMouthPosition::update()]")
+
+        # Get the inputs from the blackboard
+        face_detection_msg = self.blackboard_get("face_detection_msg")
+        frame_id = self.blackboard_get("frame_id")
+        position_offset = self.blackboard_get("position_offset")
+
+        # Transform the face detection result to the base frame. If the
+        # transform doesn't exist, then return failure.
+        try:
```

Comment:
Not a fan of this nested try style.

I'd put the outer try/except inside of the except of the inner try.